File: app/src/main/java/com/example/notes/data/note/Repository.java
package com.example.notes.data.note;

import com.example.notes.data.note.Note;

import java.util.ArrayList;

public interface Repository {
    int create(Note note);

    Note read(int id);

    void update(Note note);

    void delete(int id);

    ArrayList<Note> getAll();
}

File: app/src/main/java/com/example/notes/domain/Repository.kt
package com.example.notes.domain

import io.reactivex.rxjava3.core.Single

interface Repository {
    fun getAll(): Single<List<Note>>
    fun insert(note: Note): Single<Unit>
    fun update(note: Note): Single<Unit>
    fun delete(note: Note): Single<Unit>
}